torax/sources/source.py

```diff
@@ -441,12 +446,15 @@ def get_source_profiles(
     core_profiles: state.CoreProfiles | None,
     model_func: SourceProfileFunction,
     formula: SourceProfileFunction,
+    prescribed_values: chex.Array | None,
```

Comment:
I think the function below would fail if `prescribed_values` was `None`. And I dont think you ever pass in None, anyways. So go ahead and just have this be a `chex.Array`. Also, same point on naming from the other comment - add "face" to the name.

Reply:
Good point about the array.

Reply:
I think this one actually should be cell rather than face - the `formulas` and `models` both compute source terms on the cell grid, and before `get_source_profiles` is called the `prescribed_values_face` (on the right hand side) are interpolated onto the cell grid (left hand side) [[link](https://github.com/theo-brown/torax/blob/fbb0837ba2693bcd645e39a406efc95ea3f3cdfe/torax/sources/source.py#L221)].

Reply:
Actually, after doing a bit more digging, it might not be that simple. `get_source_profiles` is used to compute profiles on both the cell and face grids:

[Cell example, `fusion_heat_source.py`](https://github.com/google-deepmind/torax/blob/ef5a49ddf87de18dabc172ae57fd8fec4f6f2e89/torax/sources/fusion_heat_source.py#L118)
[Face example, `external_current_source.py`](https://github.com/google-deepmind/torax/blob/ef5a49ddf87de18dabc172ae57fd8fec4f6f2e89/torax/sources/external_current_source.py#L208C22-L208C41)

Reply:
Not sure whether this is intentional/something that should be continued, or whether it's a quirk of the `jext` stuff. There are a few things that are only used in `jext` that are not used elsewhere. Is `jext` also due a refactor, to make it more flexible?

Reply:
See comment above. Unless I'm confused, the input into `prescribed_values` on the config level can be on any user-provided grid (as long as it contains rhon=0 and rhon=1), and the `get_value` eventually provides the face grid interpolation that can then easily also be converted to the cell grid

Reply:
Based on @jcitrin's comment, I'm leaving it as `prescribed_values` unless there are any further objections.